### PR TITLE
[FLINK-33506] Add support for jdk17 for AWS connectors

### PR DESCRIPTION
**Pull request by snuyanzin**
## Purpose of the change

The PR adds support for jdk17, it also adds java 17 to ci build matrix

## Verifying this change
maven

This change is already covered by existing tests + java 17 added to ci

## Significant changes

- [X] Dependencies have been added or upgraded

### CODE REVIEW

**Comment by snuyanzin** [comment]
@dannycranmer could you have a look here please?
It will allow to check PRs against java 17

File: flink-connector-aws/flink-connector-dynamodb/pom.xml [hunk 1]
```diff
@@ -34,6 +34,9 @@ under the License.
 
     <packaging>jar</packaging>
 
+    <properties>
+        <flink.connector.module.config><!-- Required by DynamoDbSinkITCase --> --add-opens=java.base/java.util=ALL-UNNAMED</flink.connector.module.config>
+    </properties>
```
Comment (z3d1k) [inline flink-connector-aws/flink-connector-dynamodb/pom.xml:39]: `DynamoDbSinkITCase` also requires `java.base/java.lang` to be open
```
[ERROR] org.apache.flink.connector.dynamodb.sink.DynamoDbSinkITCase.testRandomDataSuccessfullyWritten  Time elapsed: 0.045 s  <<< ERROR!
java.lang.reflect.InaccessibleObjectException: Unable to make field private final byte[] java.lang.String.value accessible: module java.base does not "opens java.lang" to unnamed module @365c30cc
        at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:354)
        at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
        at java.base/java.lang.reflect.Field.checkCanSetAccessible(Field.java:178)
        at java.base/java.lang.reflect.Field.setAccessible(Field.java:172)
        at org.apache.flink.api.java.ClosureCleaner.clean(ClosureCleaner.java:106)
        at org.apache.flink.api.java.ClosureCleaner.clean(ClosureCleaner.java:132)
        at org.apache.flink.api.java.ClosureCleaner.clean(ClosureCleaner.java:69)
        at org.apache.flink.streaming.api.environment.StreamExecutionEnvironment.clean(StreamExecutionEnvironment.java:2317)
        at org.apache.flink.streaming.api.datastream.DataStream.clean(DataStream.java:202)
        at org.apache.flink.streaming.api.datastream.DataStream.map(DataStream.java:576)
        at org.apache.flink.connector.dynamodb.sink.DynamoDbSinkITCase.getRandomDataGenerator(DynamoDbSinkITCase.java:230)
        at org.apache.flink.connector.dynamodb.sink.DynamoDbSinkITCase.testRandomDataSuccessfullyWritten(DynamoDbSinkITCase.java:230)
```

**Review by z3d1k — COMMENTED** [review]
Thanks for looking into this @snuyanzin

File: flink-connector-aws/flink-connector-dynamodb/pom.xml (hunk 1 above)
Comment (snuyanzin) [reply flink-connector-aws/flink-connector-dynamodb/pom.xml]: sorry it seems I missed it....

now fixed
please have a look once you have time

**Review by z3d1k — APPROVED** [review]
LGTM, thanks @snuyanzin!

**Review by dannycranmer — APPROVED** [review]
LGTM, thanks! @snuyanzin  :D

**Pull request merged**